cargo-shuttle/src/args.rs

```diff
@@ -38,4 +45,25 @@ pub struct AuthArgs {
 pub struct DeployArgs {
     #[structopt(long, about = "allow dirty working directories to be packaged")]
     pub allow_dirty: bool,
+    #[structopt(flatten)]
+    pub project_args: ProjectArgs,
+}
+
+fn parse_working_directory(working_directory: &OsStr) -> Result<PathBuf, OsString> {
+    canonicalize(working_directory)
+        .map_err(|e| format!("could not turn {working_directory:?} into a real path: {e}").into())
+}
+
+// Common args for subcommands that deal with projects.
+#[derive(StructOpt)]
+pub struct ProjectArgs {
```

Comment:
This can now be combined with the recently created [`Args`](https://github.com/shuttle-hq/shuttle/blob/6a13ff0525517a36043ba6f0bbfa0cd0f3047559/cargo-shuttle/src/args.rs#L12)

Reply:
rebased and force-pushed.

I added this to Args as 
```rust
    #[structopt(flatten)]
    pub project_args: ProjectArgs,
```
because it was nice to be able to pass it as a single arg into `RequestContext::load_local()`. This is the simplest approach, but has slightly awkward consequences in the UI (see below).

I also marked `name` and `working_directory` as `global = true`. This means that they will apply to all subcommands (unfortunately, this means you can do `cargo shuttle login --name=myproject` even though `login` doesn't know anything about projects).

(I did not touch api_url, because api_url feels like a kind-of hidden feature, that you only use for local dev, and probably shouldn't be in everyone's face)

I'm really tempted to put ProjectArgs back into the payloads of each of the project-related sub-command enums (this is where I originally put them). This does not allow them to be specified at the top level (`cargo shuttle --name=myproject status`). It would also be possible to put ProjectArgs in Args and also each of the subcommands, and merge them, but this feels like a maintenance headache.

Reply:
Ah, I see what you mean about `login` not using them. For now, I think it's fine to keep the code simpler.

I never knew about `global = true` so learned something new :smile: . The `api_url` will be used for self-deploys soon. But it can remain hidden for now.

Let's avoid an args merging maintenance headache :sweat_smile: